### PR TITLE
[v1.15.x] CNI related fixes

### PR DESCRIPTION
**Pull request by waynz0r**
| Q               | A
| --------------- | ---
| Bug fix?        | yes
| New feature?    | no
| API breaks?     | no
| Deprecations?   | no
| Related tickets | 
| License         | Apache 2.0

### What's in this PR?

CNI component related fixes to properly support multiple CNI daemonsets for each control plane on one cluster.

- [x] Implementation tested

### CODE REVIEW

**Review by Laci21 — APPROVED** [review]
lgtm, thank you very much, @waynz0r!

File: internal/assets/manifests/istio-cni/templates/daemonset.yaml [hunk 1]
```diff
@@ -71,6 +71,8 @@ spec:
                   key: cni_network_config
```
Comment (tiswanso) [inline internal/assets/manifests/istio-cni/templates/daemonset.yaml:71]: edit: This comment is for CNI_CONF_NAME env var not being modified in this changeset

Will try to followup more closely on this but I was assuming the filename would need to be `name-with-revision` as well.  In the chat test info you showed I saw the filename had rev appended so probably this works with this changeset so I'll try to grok whether there's any missing conditions.

Comment (waynz0r) [reply internal/assets/manifests/istio-cni/templates/daemonset.yaml:71]: The CNI_CONF_NAME value is not generated, but it specified by the user in the icp spec, so it is up to the user the set it properly.

Comment (waynz0r) [reply internal/assets/manifests/istio-cni/templates/daemonset.yaml:71]: Also, it is the actual filename of the config on the disk, it is not the `name` property inside the config.

**Review by tiswanso — APPROVED** [review]

**Pull request merged**